eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -199,6 +236,17 @@ instance CC.Crypto crypto => ToJSON (ShelleyGenesisStaking crypto) where
       [ "pools" .= sgsPools sgs,
         "stake" .= sgsStake sgs
       ]
+  toEncoding sgs =
+    -- Forced evaluation of each field allows the parent object to no longer be
+    -- referenced, which helps prevent retention of any fields that have already
+    -- been serialised.
+    let !pools = sgsPools sgs
+        !stake = sgsStake sgs
```

Comment:
Comment and strict let bindings are unnecessary, both of these fields are strict. Can you switch them to NamedFieldPuns?